### ISSUE TITLE
Failure to create compute instance: both identity then key-based access

### ISSUE THREAD

**Issue by IainD925** [Azure-Samples/ai-studio-in-a-box#7]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>Deployed using Identity, received error reported previously in issue #2  

```
StorageMountError: The specified Azure ML Compute Instance vmai-inabox-r6b setup failed with error "Failed to mount storage. Details - Hint:If you're using network security group rules, make sure to allow outbound connection to ports 443, 445 for storage service tag. For more details please visit aka.ms/AMLComputeVnet ". Please delete and try to recreate. If the problem persists, please follow up with Azure Support.
```
Wiped deployment then tried key-based which also failed - log below.  Partly tied to issue #6

### Any log messages given by the failure
>azd up
? Select an Azure Subscription to use:  3. Microsoft Azure Sponsorship Subscription 3 (abcd123-735a-4fac-91a2-677d3e1ad96b)
? Select an Azure location to use: 45. (US) East US 2 (eastus2)
? Enter a value for the 'deploySearch' infrastructure parameter: True
? Enter a value for the 'publicNetworkAccess' infrastructure parameter: Enabled
? Enter a value for the 'systemDatastoresAuthMode' infrastructure parameter: accessKey

Packaging services (azd package)

Provisioning Azure resources (azd provision)
Provisioning Azure resources can take some time.

Subscription: Microsoft Azure Sponsorship Subscription 3 (abcd123-735a-4fac-91a2-677d3e1ad96b)
Location: East US 2

  You can view detailed progress in the Azure Portal:
  https://portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2Fabcd123-735a-4fac-91a2-677d3e1ad96b%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Faiinabox-1722593420

  (✓) Done: Resource group: rg-aiinabox
  (✓) Done: Virtual Network: vnet-aiinabox-whq
  (✓) Done: Storage account: staiinaboxwhq
  (✓) Done: Key Vault: kv-aiinabox-whq
  (✓) Done: Private Endpoint: pl-staiinaboxwhq
  (✓) Done: Private Endpoint: pl-kv-aiinabox-whq
  (✓) Done: Search service: srch-aiinabox-whq
  (✓) Done: Private Endpoint: pl-srch-aiinabox-whq
  (✓) Done: Cognitive Service: cog-aiinabox-whq
  (✓) Done: Private Endpoint: pl-cog-aiinabox-whq
  (✓) Done: Private Endpoint: pl-oai-cog-aiinabox-whq
  (✓) Done: Machine Learning Workspace: cog-hub-aiinabox-whq
  (✓) Done: Machine Learning Workspace: cog-ai-prj-aiinabox-whq
  (x) Failed: Private Endpoint: pl-cog-hub-aiinabox-whq
  |       | Creating/Updating resources
ERROR: error executing step command 'provision': deployment failed: error deploying infrastructure: deploying to subscription:

Deployment Error Details:
InternalServerError: An error occurred.

TraceID: b3b6af0649aa142177c0bb987c89a37d
PS C:\ai-studio-in-a-box> azd up

Packaging services (azd package)

Provisioning Azure resources (azd provision)
Provisioning Azure resources can take some time.

Subscription: Microsoft Azure Sponsorship Subscription 3 (abcd123-735a-4fac-91a2-677d3e1ad96b)
Location: East US 2

  You can view detailed progress in the Azure Portal:
  https://portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2Fabcd123-735a-4fac-91a2-677d3e1ad96b%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Faiinabox-1722596046

  (✓) Done: Resource group: rg-aiinabox
  (✓) Done: Virtual Network: vnet-aiinabox-whq
  (✓) Done: Search service: srch-aiinabox-whq
  (✓) Done: Private Endpoint: pl-srch-aiinabox-whq
  (✓) Done: Key Vault: kv-aiinabox-whq
  (✓) Done: Storage account: staiinaboxwhq
  (✓) Done: Private Endpoint: pl-kv-aiinabox-whq
  (✓) Done: Private Endpoint: pl-staiinaboxwhq
  (✓) Done: Cognitive Service: cog-aiinabox-whq
  (✓) Done: Private Endpoint: pl-cog-aiinabox-whq
  (✓) Done: Private Endpoint: pl-oai-cog-aiinabox-whq
  (✓) Done: Machine Learning Workspace: cog-hub-aiinabox-whq
  (x) Failed: Machine Learning Workspace: cog-ai-prj-aiinabox-whq
  (✓) Done: Private Endpoint: pl-cog-hub-aiinabox-whq
  |      =| Creating/Updating resources
ERROR: error executing step command 'provision': deployment failed: error deploying infrastructure: deploying to subscription:

Deployment Error Details:
ValidationError: Managed network cannot be disabled once enabled.
BadRequest: When updating a shared private link resource, only 'requestMessage' property is allowed to be modified RequestId: 5c8f01b0-ba52-db4f-8ddc-416ff27fa2be

TraceID: 5c8f01b0ba52db4f8ddc416ff27fa2be
PS C:\ai-studio-in-a-box> azd up

Packaging services (azd package)

Provisioning Azure resources (azd provision)
Provisioning Azure resources can take some time.

Subscription: Microsoft Azure Sponsorship Subscription 3 (abcd123-735a-4fac-91a2-677d3e1ad96b)
Location: East US 2

  You can view detailed progress in the Azure Portal:
  https://portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2Fabcd123-735a-4fac-91a2-677d3e1ad96b%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Faiinabox-1722596932

  (✓) Done: Resource group: rg-aiinabox
  (✓) Done: Virtual Network: vnet-aiinabox-whq
  (✓) Done: Search service: srch-aiinabox-whq
  (✓) Done: Key Vault: kv-aiinabox-whq
  (✓) Done: Storage account: staiinaboxwhq
  (✓) Done: Private Endpoint: pl-srch-aiinabox-whq
  (✓) Done: Private Endpoint: pl-kv-aiinabox-whq
  (✓) Done: Private Endpoint: pl-staiinaboxwhq
  (✓) Done: Cognitive Service: cog-aiinabox-whq
  (x) Failed: Private Endpoint: pl-oai-cog-aiinabox-whq
  (x) Failed: Private Endpoint: pl-cog-aiinabox-whq
  |=      | Creating/Updating resources
ERROR: error executing step command 'provision': deployment failed: error deploying infrastructure: deploying to subscription:

Deployment Error Details:
AccountProvisioningStateInvalid: Call to Microsoft.CognitiveServices/accounts failed. Error message: Account /subscriptions/abcd123-735a-4fac-91a2-677d3e1ad96b/resourceGroups/rg-aiinabox/providers/Microsoft.CognitiveServices/accounts/cog-aiinabox-whq in state Accepted
AccountProvisioningStateInvalid: Call to Microsoft.CognitiveServices/accounts failed. Error message: Account /subscriptions/abcd123-735a-4fac-91a2-677d3e1ad96b/resourceGroups/rg-aiinabox/providers/Microsoft.CognitiveServices/accounts/cog-aiinabox-whq in state Accepted

TraceID: 73c0d9ffb67cc508e3eec70404b46d41
PS C:\ai-studio-in-a-box> azd up

Packaging services (azd package)

Provisioning Azure resources (azd provision)
Provisioning Azure resources can take some time.

Subscription: Microsoft Azure Sponsorship Subscription 3 (abcd123-735a-4fac-91a2-677d3e1ad96b)
Location: East US 2

  You can view detailed progress in the Azure Portal:
  https://portal.azure.com/#view/HubsExtension/DeploymentDetailsBlade/~/overview/id/%2Fsubscriptions%2Fabcd123-735a-4fac-91a2-677d3e1ad96b%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Faiinabox-1722597431

  (✓) Done: Resource group: rg-dns
  |    ===| Creating/Updating resources
ERROR: error executing step command 'provision': deployment failed: error deploying infrastructure: deploying to subscription:

Deployment Error Details:
ValidationError: Managed network cannot be disabled once enabled.
BadRequest: When updating a shared private link resource, only 'requestMessage' property is allowed to be modified RequestId: b52e60dc-69fa-0c47-2779-38fa72b29d89

TraceID: b52e60dc69fa0c47277938fa72b29d89

### Expected/desired behavior
>

### OS and Version?
> Windows 10.

### Versions
>https://github.com/Azure-Samples/ai-studio-in-a-box/commit/f6754c3b85a83b5bdae504b8227a099a4c80b532

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by MarcoABCardoso** [comment]
Hello @IainD925!

Your first issue is known as well - Compute Instances have an issue when mounting storage with identity-based access. There is a fix being worked on for this issue, but I don't think it's fully rolled out yet.

The workaround is described in the [FAQ](https://github.com/Azure-Samples/ai-studio-in-a-box?tab=readme-ov-file#faq):

> Compute instance setup fails with StorageMountError:

> StorageMountError: The specified Azure ML Compute Instance ******** setup failed with error "Failed to mount storage. Details - Hint:If you're using network security group rules, make sure to allow outbound connection to ports 443, 445 for storage service tag. For more details please visit aka.ms/AMLComputeVnet"

> Compute instances are unable to mount for the first time with identity-based access. If you require compute instances, first deploy with key-based access, create one compute, and later switch to identity based. After this process, you will be able to create new computes normally.

I believe your subsequent deploys failed due to #6 - Shared Private Links and AI Project fail to re-deploy.

In short:
- Deploy normally with key-based access: compute should come up normally
- Deploy over the existing resources with identity-based access, disabled Shared Private Links and disabled AI Project deployment. This will update the Hub to use its identity once storage is configured.

I'll provide updates on the compute instance issue as they become available.
